specification.md

```diff
@@ -1903,10 +1903,30 @@ Following figure describes the main workflow definition blocks.
 
 The `id` property defines the unique, domain-specific workflow identifier, for example "orders", "payment", etc.
 
-The `key` property defines the unique, domain-specific workflow identifier.
-It can be used when the `id` property is auto-generated by a content-management system for example.
-In these cases, you can specify the `key` property to be the domain-specific identifier of the workflow definition.
-The `id` and `key` properties are mutually exclusive, meaning you cannot define both.
+The `key` property defines the unique, domain-specific workflow instance identifier. It is an expression used by runtimes to
+determine the unique identifier of instances against their input data.
+
+For example, consider the following definition:
+
+```yaml
+id: pay
+version: '1.0'
+specVersion: '0.8'
+key: '${ .orderId }'
+states: []
+```
+
+If this workflow is instanciated with the following data...:
+
+```yaml
+payment:
+  total: 53.21
+  currency: EUR
+  type: card
+  orderId: 1234
+```
+
+... the resulting workflow instance would get the '1234' unique identifier.
```

Comment:
@cdavernas can you also please incorporate the other comments?

- If the expression evaluated fails, the workflow should ignore it and continue the execution (how this is passed to the end user is an implementation detail) (https://github.com/serverlessworkflow/specification/issues/584#issuecomment-1026777370)
- The key is immutable once the workflow is created (https://github.com/serverlessworkflow/specification/issues/584#issuecomment-1026752612) and (https://github.com/serverlessworkflow/specification/issues/584#issuecomment-1026776835)

Reply:
Done! @ricardozanini Is it good enough for you?